src/packages/cli/commands/generate.js

```diff
@@ -49,13 +50,18 @@ export async function generateType(type, name, cwd, attrs = []) {
     case 'controller':
       data = controllerTemplate(name, attrs);
       break;
+
+    case 'middleware':
+      data = middlewareTemplate(name);
+      break;
   }
 
   if (type === 'model') {
```

Comment:
Ahh here's the culprit. We should probably replace that with a regexp `/^(model|middleware)$/.test(type)`.